modules/sw360/sw360-client/src/main/java/org/eclipse/sw360/antenna/sw360/client/adapter/SW360LicenseClientAdapterAsyncImpl.java

```diff
@@ -36,20 +37,22 @@ public SW360LicenseClient getLicenseClient() {
     }
 
     @Override
-    public CompletableFuture<Boolean> isLicenseOfArtifactAvailable(String license) {
-        return getLicenseClient().getLicenses()
-                .thenApply(sw360Licenses -> sw360Licenses.stream()
-                        .map(SW360SparseLicense::getShortName)
-                        .anyMatch(n -> n.equals(license)));
+    public CompletableFuture<List<SW360SparseLicense>> getLicenses() {
+        return getLicenseClient().getLicenses();
     }
 
     @Override
-    public CompletableFuture<Optional<SW360License>> getSW360LicenseByAntennaLicense(String license) {
+    public CompletableFuture<Optional<SW360License>> getLicenseByName(String license) {
         return optionalFuture(getLicenseClient().getLicenseByName(license));
     }
 
     @Override
-    public CompletableFuture<Optional<SW360License>> getLicenseDetails(SW360SparseLicense sparseLicense) {
-        return optionalFuture(getLicenseClient().getLicenseByName(sparseLicense.getShortName()));
+    public CompletableFuture<SW360License> enrichSparseLicense(SW360SparseLicense sparseLicense) {
+        return getLicenseClient().getLicenseByName(sparseLicense.getShortName());
+    }
+
+    @Override
+    public CompletableFuture<SW360License> createLicense(SW360License license) {
+        return getLicenseClient().createLicense(license);
```

Comment:
Whilst this is currently only called in tests, when using it we must be aware that there must be a check whether the license contains a fullname (the only required attribute that needs to be set in a SW360 License object). Currently we most of the time use the short name attribute in Antenna functionalities. The full name is currently only ever set in tests.

Reply:
Additionally, whilst I agree that this method should be included into the sw360-client library, should we ever use it in an antenna context, we need to be careful not to write bogus licenses into our instance, as has happened before.

Reply:
Regarding the fullname attribute: At least SW360 rejects requests to add a license without a full name. We could add a check already in the client library to prevent server calls that will fail anyway. I am, however, not sure how far we should go with duplicating validation logic. There is always the risk that the client validation gets out of sync with the server validation. Letting SW360 reject the call has the advantage that the server sends an official error message, which currently looks like this:
```
{
    "status": 500,
    "error": "Internal Server Error",
    "message": "Required field 'fullname' was not present! Struct: License(id:TST-3, type:license, shortname:TST-3, fullname:null, GPLv2Compat:UNDEFINED, GPLv3Compat:UNDEFINED, text:License to test)",
    "timestamp": "2020-07-10T07:17:41.045Z"
}
```

I am, however, not sure whether this message is actually made available by the client library.

Regarding the creation of licenses from Antenna, this is a good point. I assume that Antenna will use the licenses in SW360 as a read-only source; otherwise, it has to be defined exactly under which conditions a new license would have to be created.

Reply:
In my opinion the message given by SW360 looks good. I see two options:
- if the message isn't given to the user (because of error handling or sth) we should change it so that it is displayed
- make our own check and throw an error message

However to the 2nd option: I agree that we should avoid duplicated validations

Reply:
I have created an issue to make the error message from the server available: https://github.com/eclipse/antenna/issues/558